mopro-ffi/src/circom/mod.rs

```diff
@@ -175,22 +166,99 @@ pub fn generate_circom_proof_wtns(
 pub fn verify_circom_proof(
     proof_lib: ProofLib,
     zkey_path: String,
-    proof: Vec<u8>,
-    public_inputs: Vec<u8>,
+    proof_result: CircomProofResult,
 ) -> Result<bool> {
-    // TODO fix this workaround, change `public_inputs: Vec<u8>` to `public_inputs: Vec<String>`
-    let mut header_reader = ZkeyHeaderReader::new(&zkey_path);
-    header_reader.read();
-    let public_inputs = if header_reader.r == BigUint::from(ark_bn254::Fr::MODULUS) {
-        serialization::deserialize_inputs::<Bn254>(public_inputs).into()
-    } else if header_reader.r == BigUint::from(ark_bls12_381::Fr::MODULUS) {
-        serialization::deserialize_inputs::<Bls12_381>(public_inputs).into()
-    } else {
-        // unknown curve
-        bail!("unknown curve detected in zkey")
-    };
+    CircomProver::verify(
+        proof_lib,
+        circom_prover::prover::CircomProof {
+            proof: proof_result.proof.into(),
+            pub_inputs: proof_result.inputs.into(),
+        },
+        zkey_path,
+    )
+}
+
+//
+// `From` implementation for proof conversion
+//
+impl From<CircomProverProof> for CircomProof {
+    fn from(proof: CircomProverProof) -> Self {
+        CircomProof {
+            a: proof.a.into(),
+            b: proof.b.into(),
+            c: proof.c.into(),
+            protocol: proof.protocol,
+            curve: proof.curve,
+        }
+    }
+}
 
-    CircomProver::verify(proof_lib, proof, public_inputs, zkey_path)
+impl From<CircomProof> for CircomProverProof {
+    fn from(proof: CircomProof) -> Self {
+        CircomProverProof {
+            a: proof.a.into(),
+            b: proof.b.into(),
+            c: proof.c.into(),
+            protocol: proof.protocol,
+            curve: proof.curve,
+        }
+    }
+}
+
+impl From<CircomProverG1> for G1 {
+    fn from(g1: CircomProverG1) -> Self {
+        G1 {
+            x: g1.x.to_string(),
+            y: g1.y.to_string(),
+            z: Some(g1.z.to_string()),
```

Comment:
could you share where is the `z` element?
will it be some other value?

Reply:
I'm asumming (I'm not sure) still someone needs `ProofCalldata` struct so I kept as it was. 

Before this change, the definition of G1 and G2 is
```rust
pub struct G1 {
    pub x: String,
    pub y: String,
}

pub struct G2 {
    pub x: Vec<String>,
    pub y: Vec<String>,
}
```
Only have x and y.

However, we added `z` in this PR, and I think if we define two types of `G1` and `G2` (w/ and w/o `z`), it'll be confusing. So, that's the reason why I used the same structures but used `Option` for `z`.

TL;DR if it's a point in `ProofCalldata`, then `z` is `None`. If it's in `CircomProof`, then `z` is not `None` (Could be 1 or 0, 0 in some extrem case).